### PR TITLE
fn: improve UX and publish/deploy command

### LINKED ISSUE #352: Improve `fn publish` command
https://github.com/iron-io/functions/pull/351

Most likely `funcfile` struct can be trimmed down to fewer attributes.

### PR DESCRIPTION
**Pull request by ccirello**
Fixes #352
Partially addresses #364

### CODE REVIEW

**Comment by ccirello** [comment]
@seiflotfy @treeder @pedronasser

**Comment by ccirello** [comment]
@treeder @pedronasser @seiflotfy I have extensively checked and updated the documentation on this change. 

Please review, critique and merge if OK.

File: examples/hello/go/README.md [hunk 1]
```diff
@@ -9,10 +9,8 @@ fn init <YOUR_DOCKERHUB_USERNAME>/hello
 fn build
 # test it
 cat hello.payload.json | fn run
-# push it to Docker Hub
-fn push
 # Create a route to this function on IronFunctions
-fn routes create myapp /hello
+fn deploy myapp
```
Comment (treeder) [inline examples/hello/go/README.md]: I thought we were discussing that deploy would be more of a bulk operation only, for when you setup an entire app structure?  What path would be used here?

Comment (ccirello) [inline examples/hello/go/README.md]: You're right. fixing it.

File: examples/hello/ruby/README.md [hunk 2]
```diff
@@ -11,10 +11,8 @@ docker run --rm -v "$PWD":/worker -w /worker iron/ruby:dev bundle install --stan
 fn build
 # test it
 cat hello.payload.json | fn run
-# push it to Docker Hub for use with IronFunctions
-fn push
 # Create a route to this function on IronFunctions
-fn routes create myapp /hello
+fn deploy myapp
```
Comment (treeder) [inline examples/hello/ruby/README.md]: This should still be push and routes create, not deploy.

File: examples/slackbot/guppy/README.md [hunk 3]
```diff
@@ -11,10 +11,8 @@ docker run --rm -v "$PWD":/worker -w /worker iron/ruby:dev bundle install --stan
 fn build
 # test it
 cat slack.payload | fn run
-# push it to Docker Hub for use with IronFunctions
-fn push
 # Create a route to this function on IronFunctions
-fn routes create slackbot /guppy
+fn deploy slackbot
```
Comment (treeder) [inline examples/slackbot/guppy/README.md]: Same

**Pull request merged**